app/services/bulk_republisher.rb

```diff
@@ -41,11 +43,31 @@ def republish_all_documents_with_publicly_visible_editions_with_html_attachments
     republish_by_document_ids(document_ids)
   end
 
+  def republish_all_by_type(type)
+    begin
+      content_type = type.constantize
+    rescue NameError
+      raise "Unknown content type #{content_type}\nCheck the GOV.UK developer documentation for a list of acceptable document types: https://docs.publishing.service.gov.uk/manual/republishing-content.html#whitehall"
+    end
+
+    if non_editionable_content_types.include?(type)
+      republish_non_editionable_content_types(content_type.all)
+    else
+      republish_by_document_ids(content_type.pluck(:document_id))
+    end
+  end
+
 private
 
   def republish_by_document_ids(document_ids)
     document_ids.each do |document_id|
       PublishingApiDocumentRepublishingWorker.perform_async_in_queue("bulk_republishing", document_id, true)
     end
   end
+
+  def republish_non_editionable_content_types(content_type)
+    content_type.find_each do |type|
+      Whitehall::PublishingApi.bulk_republish_async(type)
+    end
```

Comment:
The difference between a "content type" and a "type" (instance) feels unclear - perhaps the enumerator variable here should be `instance` or `record`?

Reply:
Yeah, fair enough. Have renamed this to `record`